packages/profiles/__tests__/BasicProfileManager.save.test.ts

```diff
@@ -15,589 +15,589 @@ import { TestLogger } from "../../../__tests__/TestLogger";
-    }
-    expect(error).toBeDefined();
-    expect(error instanceof ImperativeError).toBe(true);
-    expect(error.message).toMatchSnapshot();
-  });
-
-  it("should detect that the dependencies are not an array", async () => {
-    const prof = new BasicProfileManager({
-      profileRootDirectory: TEST_PROFILE_ROOT_DIR,
-      typeConfigurations: ONLY_APPLE,
-      type: APPLE_PROFILE_TYPE,
-      logger: TestLogger.getTestLogger()
+        expect(error).toBeDefined();
+        expect(error instanceof ImperativeError).toBe(true);
+        expect(error.message).toMatchSnapshot();
     });
 
-    let error;
-    let response: IProfileSaved;
-    try {
-      const profile: any = {name: "bad_apple"};
-      profile.dependencies = {};
-      response = await prof.save({profile});
-    } catch (e) {
-      error = e;
-      TestLogger.info(error);
-    }
-    expect(error).toBeDefined();
-    expect(error instanceof ImperativeError).toBe(true);
-    expect(error.message).toMatchSnapshot();
-  });
-
-  it("should detect that the dependencies are present, but name is missing", async () => {
-    const prof = new BasicProfileManager({
-      profileRootDirectory: TEST_PROFILE_ROOT_DIR,
-      typeConfigurations: ONLY_APPLE,
-      type: APPLE_PROFILE_TYPE,
-      logger: TestLogger.getTestLogger()
+    it("should detect a type mismatch when saving a profile", async () => {
+        let error;
+        try {
+            const prof = new BasicProfileManager({
+                profileRootDirectory: TEST_PROFILE_ROOT_DIR,
+                typeConfigurations: ONLY_APPLE,
+                type: STRAWBERRY_PROFILE_TYPE,
+                logger: TestLogger.getTestLogger()
+            });
+            const response = await prof.save(BAD_SAMPLE_SAVE_PARMS);
+        } catch (e) {
+            error = e;
+            TestLogger.info(e);
+        }
+        expect(error).toBeDefined();
+        expect(error instanceof ImperativeError).toBe(true);
+        expect(error.message).toContain(
+            "Expect Error: Could not locate the profile type configuration for \"strawberry\" within the input configuration list passed."
+        );
     });
 
-    let error;
-    let response: IProfileSaved;
-    try {
-      const profile: any = {name: "bad_apple"};
-      profile.dependencies = [{type: STRAWBERRY_PROFILE_TYPE}];
-      response = await prof.save({profile});
-    } catch (e) {
-      error = e;
-      TestLogger.info(error);
-    }
-    expect(error).toBeDefined();
-    expect(error instanceof ImperativeError).toBe(true);
-    expect(error.message).toMatchSnapshot();
-  });
-
-  it("should detect that the dependencies are present, but type is missing", async () => {
-    const prof = new BasicProfileManager({
-      profileRootDirectory: TEST_PROFILE_ROOT_DIR,
-      typeConfigurations: ONLY_APPLE,
-      type: APPLE_PROFILE_TYPE,
-      logger: TestLogger.getTestLogger()
+    it("should detect a blank name when creating a profile", async () => {
```

Comment:
should there be a test on save for omitting name entirely? I don't think I saw one?